public/app/features/explore/Logs/LogsTable.tsx

```diff
@@ -14,38 +15,42 @@ import {
   ValueLinkConfig,
 } from '@grafana/data';
 import { config } from '@grafana/runtime';
-import { Table } from '@grafana/ui';
+import { AdHocFilterItem, Table } from '@grafana/ui';
+import { FILTER_FOR_OPERATOR, FILTER_OUT_OPERATOR } from '@grafana/ui/src/components/Table/types';
 import { separateVisibleFields } from 'app/features/logs/components/logParser';
-import { parseLogsFrame } from 'app/features/logs/logsFrame';
+import { LogsFrame, parseLogsFrame } from 'app/features/logs/logsFrame';
 
 import { getFieldLinksForExplore } from '../utils/links';
 
+import { fieldNameMeta } from './LogsTableWrap';
+
 interface Props {
   logsFrames?: DataFrame[];
   width: number;
   timeZone: string;
   splitOpen: SplitOpen;
   range: TimeRange;
   logsSortOrder: LogsSortOrder;
+  columnsWithMeta: Record<string, fieldNameMeta>;
+  height: number;
+  onClickFilterLabel?: (key: string, value: string, refId?: string) => void;
+  onClickFilterOutLabel?: (key: string, value: string, refId?: string) => void;
+  datasourceType?: string;
```

Comment:
I am not sure about this and I think we should try to figure out what fields to show based on frame itself and not based on what data source produced it.

Reply:
I think the only place we use the datasource string is on line 184
``` return field.typeInfo?.frame === 'json.RawMessage' && datasourceType === 'loki'; ```
@sven what is the `json.RawMessage` and is there a datasource agnostic approach for this field filter?

Reply:
Loki's `label` field in the dataframe is configured like this:

```
        {
          "name": "labels",
          "type": "other",
          "typeInfo": {
            "frame": "json.RawMessage"
          },
          "config": {
            "links": []
          }
        },
```

However I think we could do that for every `typeInfo.frame === 'json.RawMessage'` despite the datasource type.

Reply:
That breaks elastic tho 
<img width="1413" alt="image" src="https://github.com/grafana/grafana/assets/109082771/d628389c-f82b-43eb-97eb-df75628d3d90">

Reply:
I think we should consider 3 cases for log data frames:
- dataFrame has `other` type field with name `labels` (this is loki type of dataFrame)
- dataFrame has `other` type field with name `attributes` and `frame.meta.type ===  DataFrameType.LogLines`  (this is dataPlane type of dataFrame)
- dataFrame does not have any of these and then we will show all fields (this is how Elastic and other data sources are using it)

Reply:
Thank you @ivanahuckova. This should be implemented in the last commit, and this appears to be working without checking the datasource type!!
We ready to merge now? :P